### PR TITLE
#1399. [Records] Type annotations and record expressions tests updated

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Records/record_type_annotations_A07_t02.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+/// the getter for the first positional field.
+///
+/// @description Checks that it is no error if a record type has a field name
+/// that doesn't collide with the synthesized getter name of a positional field
+/// @author sgrekhov22@gmail.com
+/// @issue 498838
+
+// SharedOptions=--enable-experiment=records
+
+typedef R1 = (int $5, {String s});
```
Comment (chloestefantsova) [inline LanguageFeatures/Records/record_type_annotations_A07_t02.dart:32]: Similarly to the above, the compiler behavior for this test case is not yet defined. And same suggestion: either mark it as such or postpone adding this test until the language design decision is made.

/cc @lrhn

**Review by chloestefantsova — APPROVED** [review]
LGTM with some suggestions.

File: LanguageFeatures/Records/record_expressions_A07_t01.dart [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+/// A field name that collides with the synthesized getter name of a positional
+/// field. For example: ('pos', $0: 'named') since the named field '$0' collides
+/// with the getter for the first positional field.
+///
+/// @description Checks that record expression may be constant
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=records
+
+const r1 = const (3.14, name: "pi");
```
[the remark this answers is not in the document]
Comment (lrhn) [reply LanguageFeatures/Records/record_expressions_A07_t01.dart:33]: The grammar (https://github.com/dart-lang/language/blob/master/working/0546-patterns/records-feature-specification.md#record-expressions) allows `const (1, 2)`, so I guess we will want to allow it as a constant expression as well. At least it's shorter to write `const ([1], [2])` than `(const [1], const [2])`, even if the result is the same.

[the remark this answers is not in the document]
Comment (chloestefantsova) [reply LanguageFeatures/Records/record_expressions_A07_t01.dart:33]: Got it. Thanks for clarifying that, Lasse!

File: LanguageFeatures/Records/record_expressions_A05_t02.dart [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+/// with the getter for the first positional field.
+///
+/// @description Checks that it is no error if a record has a field name that
+/// doesn't collides with the synthesized getter name of a positional field
+/// @author sgrekhov22@gmail.com
+/// @issue 49883
+
+// SharedOptions=--enable-experiment=records
+
+Record foo() => (42, $100: "Lily was here");
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Records/record_expressions_A05_t02.dart:35]: There is issue number in the header of this test
```
/// @issue 49883
```
I'm also keeping my eye on the discussion an will update the tests if neede

[the remark this answers is not in the document]
Comment (chloestefantsova) [reply LanguageFeatures/Records/record_expressions_A05_t02.dart:35]: Sounds good.

**Pull request merged**